turtlesim/CMakeLists.txt

```diff
@@ -1,55 +1,110 @@
+
+find_package(rosidl_default_generators REQUIRED)
 find_package(Qt5Widgets REQUIRED)
-find_package(Boost REQUIRED COMPONENTS thread)
 
-include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
-link_directories(${catkin_LIBRARY_DIRS})
+rosidl_generate_interfaces(${PROJECT_NAME}
+  "msg/Color.msg"
+  "msg/Pose.msg"
 
-add_message_files(DIRECTORY msg FILES
-                  Color.msg Pose.msg)
-add_service_files(DIRECTORY srv FILES
-                  Kill.srv
-                  SetPen.srv
-                  Spawn.srv
-                  TeleportAbsolute.srv
-                  TeleportRelative.srv)
-generate_messages(DEPENDENCIES geometry_msgs std_msgs std_srvs)
+  "srv/Kill.srv"
+  "srv/SetPen.srv"
+  "srv/Spawn.srv"
+  "srv/TeleportAbsolute.srv"
+  "srv/TeleportRelative.srv"
+)
 
-catkin_package(CATKIN_DEPENDS geometry_msgs message_runtime std_msgs std_srvs)
+include_directories(
+  include 
+  ${QT_INCLUDES}
+)
 
 set(turtlesim_node_SRCS
   src/turtlesim.cpp
   src/turtle.cpp
   src/turtle_frame.cpp
 )
+
 set(turtlesim_node_HDRS
   include/turtlesim/turtle_frame.h
 )
 
-qt5_wrap_cpp(turtlesim_node_MOCS ${turtlesim_node_HDRS})
+set(dependencies
+  "rclcpp"
+  "ament_index_cpp"
+  "geometry_msgs"
+  "std_msgs"
+  "std_srvs"
+  "Qt5Widgets"
+)
 
+qt5_wrap_cpp(turtlesim_node_MOCS ${turtlesim_node_HDRS})
 
 add_executable(turtlesim_node ${turtlesim_node_SRCS} ${turtlesim_node_MOCS})
-target_link_libraries(turtlesim_node Qt5::Widgets ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-add_dependencies(turtlesim_node turtlesim_gencpp)
+target_link_libraries(turtlesim_node ${QT_LIBRARIES})
+ament_target_dependencies(turtlesim_node ${dependencies})
+
+rosidl_target_interfaces(turtlesim_node
+  ${PROJECT_NAME} "rosidl_typesupport_cpp"
+)
 
 add_executable(turtle_teleop_key tutorials/teleop_turtle_key.cpp)
-target_link_libraries(turtle_teleop_key ${catkin_LIBRARIES})
-add_dependencies(turtle_teleop_key turtlesim_gencpp)
+ament_target_dependencies(turtle_teleop_key ${dependencies})
+
 add_executable(draw_square tutorials/draw_square.cpp)
-target_link_libraries(draw_square ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-add_dependencies(draw_square turtlesim_gencpp)
+ament_target_dependencies(draw_square ${dependencies})
+
+rosidl_target_interfaces(draw_square
+  ${PROJECT_NAME} "rosidl_typesupport_cpp"
+)
+
 add_executable(mimic tutorials/mimic.cpp)
-target_link_libraries(mimic ${catkin_LIBRARIES})
-add_dependencies(mimic turtlesim_gencpp)
+ament_target_dependencies(mimic ${dependencies})
+
+rosidl_target_interfaces(mimic
+  ${PROJECT_NAME} "rosidl_typesupport_cpp"
+)
+
+install(TARGETS 
+  turtlesim_node 
+  turtle_teleop_key 
+  draw_square 
+  mimic
+  DESTINATION lib/${PROJECT_NAME}
+) 
+
+install(DIRECTORY 
+  images
+  launch
+  DESTINATION share/${PROJECT_NAME}
+  # FILES_MATCHING PATTERN "*.png" PATTERN "*.svg"
+)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
```

Comment:
Using the ROS 2 linters might no be feasible if we want the code to stay close to how it is formatted on the ROS 1 branch.